engines/cryomni3d/image/codecs/hlz.cpp

```diff
@@ -0,0 +1,140 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "cryomni3d/image/codecs/hlz.h"
+
+#include "common/stream.h"
+#include "common/textconsole.h"
+#include "graphics/surface.h"
+
+namespace Image {
+
+HLZDecoder::HLZDecoder(int width, int height) : Codec(),
+	_width(width), _height(height), _surface(nullptr) {
+}
+
+HLZDecoder::~HLZDecoder() {
+	if (_surface) {
+		_surface->free();
+		delete _surface;
+	}
+}
+
+const Graphics::Surface *HLZDecoder::decodeFrame(Common::SeekableReadStream &stream) {
+	if (!_surface) {
+		_surface = new Graphics::Surface();
+	}
+
+	_surface->create(_width, _height, Graphics::PixelFormat::createFormatCLUT8());
+
+	byte *dst = (byte *)_surface->getPixels();
+	decodeFrameInPlace(stream, -1, dst);
+
+	return _surface;
+}
+
+Graphics::PixelFormat HLZDecoder::getPixelFormat() const {
+	return Graphics::PixelFormat::createFormatCLUT8();
+}
+
+static inline bool getReg(Common::SeekableReadStream &stream, uint32 *size, uint32 *reg,
```

Comment:
Why static?

Reply:
It's a function not part of the the class much like a macro. I put it static to not populate global symbols and to avoid name clashes (it's local to current cpp file).